cyipopt/scipy_interface.py

```diff
@@ -604,21 +606,54 @@ def minimize_ipopt(fun,
                           njev=problem.njev,
                           nit=problem.nit)
 
+
 def _minimize_ipopt_iv(fun, x0, args, kwargs, method, jac, hess, hessp,
                        bounds, constraints, tol, callback, options):
     # basic input validation for minimize_ipopt that is not included in
     # IpoptProblemWrapper
-
-    x0 = np.asarray(x0)[()]
+    x0 = np.atleast_1d(x0)
     if not np.issubdtype(x0.dtype, np.number):
         raise ValueError('`x0` must be a numeric array.')
 
     # `method` does not need input validation. If `method is not None`, it is
     # passed to `scipy.optimize.minimize`, which raises a readable error if
     # the value isn't recognized.
 
-    # TODO: add input validation for `bounds` when adding
-    #  support for instances of new-style constraints (e.g. `Bounds`)
+    # Handle bounds that are either sequences (of sequences) or instances of
```

Comment:
I need to add to the documentation that `Bounds`, `LinearConstraints`, and `NonlinearConstraints` are now supported.